### PR TITLE
Installation scripts should allow for IDE build, also in Debug

### PR DESCRIPTION
**Pull request by gerhardol**
The current build scripts have a number of hardcoding and requires a number of manual updates.
This is a step to simplify the build process and make it easier for users to test creating Setup

 * Download and build the binaries when building wix project
 * Use variables in product.wxs, no Mono in VS*cmd, use Configuration variables, remove c++ projects. It is now possible to build setup in Debug
 * Use hMSBuild to build, common 2017/2015 detection of msbuild

Second step of #4069 
This is not making much difference in itself, it is a preparation for future changes

What did I do to test the code and ensure quality:
 - Build on Appveyor
 - Manual builds

Has been tested on (remove any that don't apply):
 - Windows 10

### CODE REVIEW

File: Setup/MakeMonoArchive.cmd [hunk 1]
```diff
@@ -1,43 +1,44 @@
 @echo off
```
Comment (RussKie) [inline Setup/MakeMonoArchive.cmd]: I would also rename this to something like `MakePortableArchive.cmd`

File: appveyor.yml [hunk 2]
```diff
@@ -34,4 +34,6 @@ build_script:
     Write-Output "Platform: $env:IdeVersion"
     & Setup\BuildInstallers.$env:IdeVersion.cmd
     if ($LastExitCode -ne 0) { $host.SetShouldExit($LastExitCode) }
+    #Get-ChildItem Setup\GitExtensions-*.msi | % { Push-AppveyorArtifact $_.FullName -FileName $_.Name }
```
Comment (RussKie) [inline appveyor.yml]: please remove if it is not needed

Comment (gerhardol) [reply appveyor.yml]: I would like to provide an installer too. However, version should be increased too. 
I suggest adding a comment for now

**Comment by RussKie** [comment]
We don't want to make an installer universally available.

File: Setup/MakeMonoArchive.cmd (hunk 1 above)
Comment (gerhardol) [reply Setup/MakeMonoArchive.cmd]: Suggest this for a separate issue/PR

Comment (RussKie) [reply Setup/MakeMonoArchive.cmd]: ok, nw

**Pull request merged**